keeping the sidecars alive with apiserver

muaazsaleem opened the pull request:
The api server may depend on other sidecars in unexpected ways, so keeping
them all alive.

This is a follow up to  https://github.com/zalando-incubator/kubernetes-on-aws/pull/3045 & https://github.com/zalando-incubator/kubernetes-on-aws/pull/2972

Signed-off-by: Muaaz Saleem <muhammad.muaaz.saleem@zalando.de>

Commit:
keeping the authnz-webhook alive with apiserver

Since we have required admission rules, requests
start failing before the api-server is gone.

Signed-off-by: Muaaz Saleem <muhammad.muaaz.saleem@zalando.de>

Commit:
Also keeping alive other sidecars

The api server may depend on other sidecars in unexpected ways, so keeping
them all alive.

Signed-off-by: Muaaz Saleem <muhammad.muaaz.saleem@zalando.de>

linki commented:
image-policy-webhook and nginx is missing on purpose?

linki commented:
For skipper you could maybe use skipper's built-in feature instead of the lifecycle hook hack? /cc @szuecs

Commit:
Missed a sidecar

Signed-off-by: Muaaz Saleem <muhammad.muaaz.saleem@zalando.de>

Commit:
Missed another sidecar

Signed-off-by: Muaaz Saleem <muhammad.muaaz.saleem@zalando.de>

szuecs commented:
@linki https://github.com/zalando/skipper/blob/master/skipper.go#L878-L891 , so this flag needs to be set (defaults to 45):
```
  -wait-for-healthcheck-interval duration
        period waiting to become unhealthy in the loadbalancer pool in front of this instance, before shutdown triggered by SIGINT or SIGTERM (default 45s)
```

Commit:
Using skipper's internal -wait-for-healthcheck-interval

Since skipper has a native flag to delay termination, integrated with
it's health endpoint, using that instead of a preStop hook.

Signed-off-by: Muaaz Saleem <muhammad.muaaz.saleem@zalando.de>

muaazsaleem commented:
👍

szuecs commented:
:+1:
